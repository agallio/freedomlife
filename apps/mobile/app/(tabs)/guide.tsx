import { useCallback, useState } from 'react'
import { Platform, useColorScheme, View } from 'react-native'
import { useFocusEffect, useRouter } from 'expo-router'
import AsyncStorage from '@react-native-async-storage/async-storage'
import * as Burnt from 'burnt'

// Components
import GuideMonthButton from '@repo/app/features/guide/components/guide-month-button'
import GuideList from '@repo/app/features/guide/components/guide-list'

// Contexts
import { useReadPassageContext } from '@repo/app/features/read/contexts/read-passage.context'

// Queries
import { useGuideByMonthQuery } from '@repo/app/hooks/use-guide-query'

// Utils
import dayjs from '@repo/app/utils/dayjs'

export default function GuideScreen() {
  const colorScheme = useColorScheme()
  const router = useRouter()
  const {
    selectedGuideMonth,
    setGuidedEnable,
    setGuidedDate,
    setGuidedSelectedPassage,
    setSelectedBibleVersion,
  } = useReadPassageContext()

  // States
  const [guidesHaveBeenRead, setGuidesHaveBeenRead] = useState<string[]>([])

  // Queries
  const queryData = useGuideByMonthQuery(selectedGuideMonth)

  // Methods
  const onGuideClick = (date: string) => {
    setGuidedEnable(true)
    setGuidedDate(date)
    setGuidedSelectedPassage('pl-1')
    setSelectedBibleVersion('tb')
    router.push('/read')
  }

  const onCheckMarkClick = (date: string) => {
    const title = 'Panduan Telah Dibaca'
    const message = dayjs(date, 'DD-MM-YYYY').format('DD MMMM YYYY')

    Burnt.toast({
      preset: 'custom',
      duration: 1.5,
      title: Platform.OS === 'ios' ? title : `${title}\n${message}`,
      message: Platform.OS === 'ios' ? message : '',
      icon: {
        ios: {
          name: 'checkmark.circle',
          color: colorScheme === 'light' ? '#047857' : '#6ee7b7',
        },
      },
    })
  }

  const getGuideHasBeenRead = (date: string) => {
    return guidesHaveBeenRead.includes(`read-${date}`)
  }

  const redirectToGuideMonthScreen = () => {
    router.push('/guide-month')
  }

  // Focus Effects (Native)
  useFocusEffect(
    useCallback(() => {
      const getAllStorageKeys = async () => {
        // Get all storage keys that has `read-` prefix then store it to state.
        // e.g. `read-01-01-2024`.
        const allStorageKeys = await AsyncStorage.getAllKeys()
        const allReadKeys = allStorageKeys.filter((i) => /^read\W/gm.test(i))
        setGuidesHaveBeenRead(allReadKeys)
      }

      getAllStorageKeys()
    }, []),
  )

  return (
    <>
      <View className="border-b border-[#e6e6e6] px-6 pb-4 pt-2 min-[744px]:px-40 md:px-52 lg:px-96 dark:border-[#374151]">
        <GuideMonthButton
          redirectToGuideMonthScreen={redirectToGuideMonthScreen}
        />
      </View>

      <GuideList
        queryData={queryData}
        getGuideHasBeenRead={getGuideHasBeenRead}
        onGuideClick={onGuideClick}
        onCheckMarkClick={onCheckMarkClick}
      />
    </>
  )
}
