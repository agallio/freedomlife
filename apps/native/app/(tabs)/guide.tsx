import GuideScreen from '@repo/app/features/guide'

export default GuideScreen
