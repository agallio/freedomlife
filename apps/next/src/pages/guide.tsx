import { useEffect, useState } from 'react'
import { NextPage } from 'next'
import Router from 'next/router'
import Head from 'next/head'
import dynamic from 'next/dynamic'
import { NextSeo } from 'next-seo'

// Components
import JumboHeader from '~/components/JumboHeader'
import PageTransition from '~/components/PageTransition'
import GuideLoading from '~/components/Guide/GuideLoading'
import GuideItem from '~/components/Guide/GuideItem'
const GuideError = dynamic(() => import('~/components/Guide/GuideError'))
const GuideMonthDialog = dynamic(
  () => import('~/components/Guide/GuideMonthDialog'),
)

// Icon Components
import ChevronDownIcon from '~/components/Icons/ChevronDownIcon'

// Context
import { useGuide } from '~/contexts/GuideContext'

// Utils
import { useGuides } from '~/utils/hooks/useFetchedGuide'
import { monthList } from '~/utils/constants'
import dayjs from '~/utils/dayjs'

const Guide: NextPage = () => {
  const [openMonthDialog, setOpenMonthDialog] = useState(false)
  const [monthNumber, setMonthNumber] = useState(dayjs().format('MM'))

  const { setGuideDate } = useGuide()

  const { data, isLoading, isError, refetch } = useGuides({
    month: monthNumber,
  })

  const toBibleWithDate = (date: string) => {
    setGuideDate(date)
    Router.push('/read')
    localStorage.setItem('in_guide', 'true')
    document.body.scrollTop = 0
    document.documentElement.scrollTop = 0
  }

  useEffect(() => {
    refetch()
  }, [monthNumber])

  return (
    <>
      <Head>
        <title>Panduan Baca — freedomlife</title>
      </Head>
      <NextSeo
        title="Panduan Baca — freedomlife"
        description="Halaman daftar panduan baca bulan ini. Anda dapat melihat panduan baca dan membaca panduan langsung dari aplikasi."
        openGraph={{
          url: 'https://freedomlife.id/guide',
          title: 'Panduan Baca — freedomlife',
          description:
            'Halaman daftar panduan baca bulan ini. Anda dapat melihat panduan baca dan membaca panduan langsung dari aplikasi.',
          site_name: 'freedomlife',
          images: [
            {
              url: 'https://freedomlife.id/images/og-guide.png',
              alt: `Tulisan dan logo 'freedomlife' disertai dengan keterangan: 'Panduan Baca Bulan Ini'`,
            },
          ],
        }}
        twitter={{
          cardType: 'summary_large_image',
        }}
      />

      <div className="mx-auto mb-20 max-w-sm p-6 sm:max-w-md sm:py-6">
        <JumboHeader />

        <button
          className="mt-2 flex w-full cursor-pointer items-center justify-between rounded-lg border bg-white px-4 py-2 shadow-md focus:ring-1 focus:ring-gray-400 dark:border-0 dark:bg-gray-600 lg:focus:ring-0"
          onClick={() => {
            document.body.style.overflow = 'hidden'
            setOpenMonthDialog(true)
          }}
        >
          <span>
            {monthList.find((item) => item.value === monthNumber)?.name || ''}
          </span>
          <ChevronDownIcon className="w-6 text-gray-500 dark:bg-gray-600 dark:text-white sm:hover:text-opacity-50" />
        </button>

        <PageTransition>
          {isError ? (
            <GuideError monthNumber={monthNumber} />
          ) : !data || isLoading ? (
            <GuideLoading />
          ) : (
            data.map((item, index) => (
              <GuideItem
                key={index}
                item={item}
                index={index}
                toBibleWithDate={toBibleWithDate}
              />
            ))
          )}
        </PageTransition>

        <GuideMonthDialog
          open={openMonthDialog}
          monthNumber={monthNumber}
          setMonthNumber={setMonthNumber}
          onClose={() => {
            document.body.style.overflow = 'visible'
            setOpenMonthDialog(false)
          }}
        />
      </div>
    </>
  )
}

export default Guide
