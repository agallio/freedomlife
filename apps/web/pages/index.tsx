// Screen
import HomeScreen from '@repo/app/features/home'

// Icon Components
import FreedomLifeIcon from '@repo/app/components/icons/freedomlife-icon'

export default function HomePage() {
  return (
    <div className="mx-auto flex w-full max-w-sm flex-col px-6 pb-28 pt-4 sm:max-w-md">
      <div className="flex pb-4">
        <FreedomLifeIcon className="w-[230px]" />
      </div>

      <HomeScreen />
    </div>
  )
}
