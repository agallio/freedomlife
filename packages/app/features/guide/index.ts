import GuideScreen from './GuideScreen'

export { GuideMonthScreen } from './components/GuideMonthScreen'

export default GuideScreen
