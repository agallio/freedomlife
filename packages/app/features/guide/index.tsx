import { useCallback, useState } from 'react'
import { FlatList, Platform, View, useColorScheme } from 'react-native'
import { useFocusEffect } from 'expo-router'
import { useRouter } from 'solito/router'
import AsyncStorage from '@react-native-async-storage/async-storage'
import * as Burnt from 'burnt'

// Components
import GuideMonthButton from './components/guide-month-button'
import GuideCard from './components/guide-card'
import GuideErrorCard from './components/guide-error-card'

// Contexts
import { useReadPassageContext } from '../read/contexts/read-passage.context'

// Queries
import { useGuideByMonthQuery } from '../../hooks/use-guide-query'

// Utils
import dayjs from '../../utils/dayjs'

export default function GuideScreen() {
  const colorScheme = useColorScheme()
  const router = useRouter()
  const {
    selectedGuideMonth,
    setGuidedEnable,
    setGuidedDate,
    setGuidedSelectedPassage,
    setSelectedBibleVersion,
  } = useReadPassageContext()

  // States
  const [guidesHaveBeenRead, setGuidesHaveBeenRead] = useState<string[]>([])

  // Queries
  const { data, isLoading, isError } = useGuideByMonthQuery(selectedGuideMonth)

  // Methods
  const onGuideClick = (date: string) => {
    setGuidedEnable(true)
    setGuidedDate(date)
    setGuidedSelectedPassage('pl-1')
    setSelectedBibleVersion('tb')
    router.push('/read')
  }

  const onCheckMarkClick = (date: string) => {
    const title = 'Panduan Telah Dibaca'
    const message = dayjs(date, 'DD-MM-YYYY').format('DD MMMM YYYY')

    Burnt.toast({
      preset: 'custom',
      title: Platform.OS === 'ios' ? title : `${title}\n${message}`,
      message: Platform.OS === 'ios' ? message : '',
      icon: {
        ios: {
          name: 'checkmark.circle',
          color: colorScheme === 'light' ? '#047857' : '#6ee7b7',
        },
      },
    })
  }

  const getGuideHasBeenRead = (date: string) => {
    return guidesHaveBeenRead.includes(`read-${date}`)
  }

  // Focus Effects (Native)
  useFocusEffect(
    useCallback(() => {
      const getAllStorageKeys = async () => {
        // Get all storage keys that has `read-` prefix then store it to state.
        // e.g. `read-01-01-2024`.
        const allStorageKeys = await AsyncStorage.getAllKeys()
        const allReadKeys = allStorageKeys.filter((i) => /^read\W/gm.test(i))
        setGuidesHaveBeenRead(allReadKeys)
      }

      getAllStorageKeys()
    }, []),
  )

  return (
    <>
      <View className="border-b border-[#e6e6e6] px-6 pb-4 pt-2 md:px-52 lg:px-96 dark:border-[#374151]">
        <GuideMonthButton />
      </View>

      {isError ? (
        <View className="gap-4 px-6 pt-4 md:px-52 lg:px-96">
          <GuideErrorCard />
        </View>
      ) : isLoading ? (
        <View className="gap-4 px-6 pb-28 pt-4 md:px-52 lg:px-96">
          {[...Array(3).keys()].map((_, i) => (
            <GuideCard key={i} isLoading />
          ))}
        </View>
      ) : (
        <FlatList
          data={data}
          keyExtractor={(_, index) => `item-${index}`}
          scrollEventThrottle={16}
          initialNumToRender={5}
          maxToRenderPerBatch={5}
          windowSize={4}
          renderItem={({ item }) => (
            <GuideCard
              item={item}
              isActive={item.date === dayjs().format('DD-MM-YYYY')}
              isRead={getGuideHasBeenRead(item.date!)}
              onGuideClick={onGuideClick}
              onCheckMarkClick={onCheckMarkClick}
            />
          )}
          contentContainerClassName="pt-4 px-6 pb-28 gap-4 md:px-52 lg:px-96"
        />
      )}
    </>
  )
}
