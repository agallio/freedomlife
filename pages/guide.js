import React, { Component } from 'react';
import Router from 'next/router';
import { connect } from 'react-redux';
import {
  Fade,
  LinearProgress,
  Card,
  CardContent,
  CircularProgress,
  Grid,
  Typography,
  ButtonBase
} from '@material-ui/core';
import moment from 'moment';
import 'moment/locale/id';

import {
  fetchGuideByMonth,
  fetchGuideByDate,
  setGuideDate,
  fetchGuide2020ByMonth
} from '../src/actions/guide';
import { fetchChapterByDate } from '../src/actions/bible';

class Guide extends Component {
  componentDidMount = () => {
    if (
      moment(this.props.guide.guideToday.date, 'DD-MM-YYYY').format('MM') !==
        moment().format('MM') ||
      this.props.guide.guideByMonth.length === 0
    ) {
      if (this.props.guide.new_2020) {
        this.props.fetchGuide2020ByMonth(
          moment().format('MM'),
          moment().format('YYYY')
        );
      } else {
        this.props.fetchGuideByMonth(
          moment().format('MM'),
          moment().format('YYYY')
        );
      }
    }
  };

  toBible = async date => {
    console.log('Clicked');
    // await this.props.setGuideDate(date);
    // await this.props.fetchGuideByDate(date);
    // await this.props.fetchChapterByDate(date);
    // Router.push('/bible');
  };

  render() {
    const {
      isFetching,
      new_2020,
      guideByMonth,
      guide2020ByMonth
    } = this.props.guide;

    return (
      <Fade in>
        <>
          {isFetching && <LinearProgress color="secondary" />}
          <div className="container" style={{ paddingBottom: 30 }}>
            <h2 className="header-title" style={{ marginTop: 35 }}>
              Panduan Baca <br /> Bulan Ini
            </h2>
            <p className="header-subtitle">
              Aplikasi panduan baca Alkitab setahun
            </p>
          </div>
          <div className="container-fluid" style={{ paddingBottom: 50 }}>
            <Card className="styled-fluid-card">
              <CardContent
                style={{
                  padding: '16px 10%',
                  height: isFetching && '80vh',
                  textAlign: isFetching && 'center',
                  paddingTop: isFetching && '20vh'
                }}
              >
                {isFetching ? (
                  <CircularProgress />
                ) : new_2020 ? (
                  guide2020ByMonth.map((item, index) => (
                    <Grid
                      container
                      key={index}
                      direction="row"
                      justify="flex-start"
                      alignItems="center"
                      spacing={4}
                    >
                      <Grid item sm={4} md={4}>
                        <ButtonBase onClick={() => this.toBible(item.date)}>
                          <div
                            className={
                              moment().format('DD-MM-YYYY') === item.date
                                ? 'guide-box-primary'
                                : 'guide-box'
                            }
                          >
                            <Typography
                              className={
                                moment().format('DD-MM-YYYY') === item.date
                                  ? 'bold-text'
                                  : 'bold-text primary'
                              }
                              style={{
                                color:
                                  moment().format('DD-MM-YYYY') === item.date &&
                                  '#fff'
                              }}
                            >
                              {moment(item.date, 'DD-MM-YYYY').format('dddd')}
                            </Typography>
                            <Typography
                              variant="h4"
                              className={
                                moment().format('DD-MM-YYYY') === item.date
                                  ? 'bold-text'
                                  : 'bold-text primary'
                              }
                              style={{
                                color:
                                  moment().format('DD-MM-YYYY') === item.date &&
                                  '#fff'
                              }}
                            >
                              {item.date.split('-')[0] || '-'}
                            </Typography>
                          </div>
                        </ButtonBase>
                      </Grid>
                      <Grid item sm={8} md={8}>
                        <Typography
                          variant="subtitle1"
                          className={
                            moment().format('DD-MM-YYYY') === item.date
                              ? 'regular-text primary'
                              : 'regular-text'
                          }
                        >
                          {item.pl_name || '-'}
                        </Typography>
                        <Typography
                          variant="subtitle1"
                          className={
                            moment().format('DD-MM-YYYY') === item.date
                              ? 'regular-text primary'
                              : 'regular-text'
                          }
                        >
                          {item.pb_name || '-'}
                        </Typography>
                        {item.alt_name && (
                          <Typography
                            variant="subtitle1"
                            className={
                              moment().format('DD-MM-YYYY') === item.date
                                ? 'regular-text primary'
                                : 'regular-text'
                            }
                          >
                            {item.alt_name || '-'}
                          </Typography>
                        )}
                      </Grid>
                    </Grid>
                  ))
                ) : (
                  guideByMonth.map((item, index) => (
                    <Grid
                      container
                      key={index}
                      direction="row"
                      justify="flex-start"
                      alignItems="center"
                      spacing={4}
                    >
                      <Grid item sm={4} md={4}>
                        <ButtonBase onClick={() => this.toBible(item.date)}>
                          <div
                            className={
                              moment().format('DD-MM-YYYY') === item.date
                                ? 'guide-box-primary'
                                : 'guide-box'
                            }
                          >
                            <Typography
                              className={
                                moment().format('DD-MM-YYYY') === item.date
                                  ? 'bold-text'
                                  : 'bold-text primary'
                              }
                              style={{
                                color:
                                  moment().format('DD-MM-YYYY') === item.date &&
                                  '#fff'
                              }}
                            >
                              {moment(item.date, 'DD-MM-YYYY').format('dddd')}
                            </Typography>
                            <Typography
                              variant="h4"
                              className={
                                moment().format('DD-MM-YYYY') === item.date
                                  ? 'bold-text'
                                  : 'bold-text primary'
                              }
                              style={{
                                color:
                                  moment().format('DD-MM-YYYY') === item.date &&
                                  '#fff'
                              }}
                            >
                              {item.date.split('-')[0] || '-'}
                            </Typography>
                          </div>
                        </ButtonBase>
                      </Grid>
                      <Grid item sm={8} md={8}>
                        <Typography
                          variant="subtitle1"
                          className={
                            moment().format('DD-MM-YYYY') === item.date
                              ? 'regular-text primary'
                              : 'regular-text'
                          }
                        >
                          {item.pl_name || '-'}
                        </Typography>
                        <Typography
                          variant="subtitle1"
                          className={
                            moment().format('DD-MM-YYYY') === item.date
                              ? 'regular-text primary'
                              : 'regular-text'
                          }
                        >
                          {item.pb1_name || '-'}
                        </Typography>
                        <Typography
                          variant="subtitle1"
                          className={
                            moment().format('DD-MM-YYYY') === item.date
                              ? 'regular-text primary'
                              : 'regular-text'
                          }
                        >
                          {item.pb2_name || '-'}
                        </Typography>
                      </Grid>
                    </Grid>
                  ))
                )}
              </CardContent>
            </Card>
          </div>
        </>
      </Fade>
    );
  }
}

const mapStateToProps = state => {
  return { guide: state.guide };
};

const mapDispatchToProps = dispatch => {
  return {
    fetchGuideByMonth: (month, year) =>
      dispatch(fetchGuideByMonth(month, year)),
    fetchChapterByDate: date => dispatch(fetchChapterByDate(date)),
    fetchGuideByDate: date => dispatch(fetchGuideByDate(date)),
    setGuideDate: date => dispatch(setGuideDate(date)),
    // 2020
    fetchGuide2020ByMonth: (month, year) =>
      dispatch(fetchGuide2020ByMonth(month, year))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Guide);
