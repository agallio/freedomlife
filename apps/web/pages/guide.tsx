import Head from 'next/head'
import { useRouter } from 'next/router'
import { NextSeo } from 'next-seo'
import * as Burnt from 'burnt'
import { CheckCircleIcon } from 'react-native-heroicons/outline'

// Components
import Drawer from '@repo/app/components/drawer'
import GuideList from '@repo/app/features/guide/components/guide-list'
import GuideMonthButton from '@repo/app/features/guide/components/guide-month-button'
import GuideMonthList from '@repo/app/features/guide/components/guide-month-list'
import { ToasterWebComponent } from '@repo/app/components/toaster-container.web'

// Icon Components
import FreedomlifeIcon from '@repo/app/components/icons/freedomlife-icon'

// Contexts
import {
  GuideModalsContextProvider,
  useGuideModalsContext,
} from '@repo/app/features/guide/contexts/guide-modals.context'
import {
  useReadPassageGeneralContext,
  useReadPassagePersistedContext,
} from '@repo/app/features/read/contexts/read-passage.context'

// Queries
import { useGuideByMonthQuery } from '@repo/app/hooks/use-guide-query'

// Utils
import dayjs from '@repo/app/utils/dayjs'

export default function GuidePage() {
  const router = useRouter()
  const { setGuidedEnabled } = useReadPassagePersistedContext()
  const selectedGuideMonth = useReadPassageGeneralContext(
    (state) => state.selectedGuideMonth,
  )
  const { setGuidedDate, setGuidedSelectedPassage, setSelectedBibleVersion } =
    useReadPassageGeneralContext((state) => state.actions)

  // Queries
  const queryData = useGuideByMonthQuery(selectedGuideMonth)

  // Methods
  const getGuideHasBeenRead = (date: string) => {
    return localStorage.getItem(`read-${date}`) !== null
  }

  const onGuideClick = (date: string) => {
    setGuidedEnabled(true)
    setGuidedDate(date)
    setGuidedSelectedPassage('pl-1')
    setSelectedBibleVersion('tb')
    router.push('/read')
  }

  const onCheckMarkClick = (date: string) => {
    Burnt.toast({
      preset: 'done',
      duration: 1.5,
      // @ts-ignore
      title: (
        <ToasterWebComponent
          icon={
            <CheckCircleIcon
              size={24}
              className="text-emerald-900 dark:text-white"
            />
          }
          title="Panduan Telah Dibaca"
          message={dayjs(date, 'DD-MM-YYYY').format('DD MMMM YYYY')}
        />
      ),
    })
  }

  return (
    <>
      <Head>
        <title>Panduan Baca — freedomlife</title>
      </Head>
      <NextSeo
        title="Panduan Baca — freedomlife"
        description="Halaman daftar panduan baca bulan ini. Anda dapat melihat panduan baca dan membaca panduan langsung dari aplikasi."
        openGraph={{
          url: 'https://freedomlife.id/guide',
          title: 'Panduan Baca — freedomlife',
          description:
            'Halaman daftar panduan baca bulan ini. Anda dapat melihat panduan baca dan membaca panduan langsung dari aplikasi.',
          site_name: 'freedomlife',
          images: [
            {
              url: 'https://freedomlife.id/images/og-guide.png',
              alt: `Tulisan dan logo 'freedomlife' disertai dengan keterangan: 'Panduan Baca Bulan Ini'`,
            },
          ],
        }}
        twitter={{
          cardType: 'summary_large_image',
        }}
      />

      <GuideModalsContextProvider>
        <div className="mx-auto flex w-full max-w-sm flex-col px-6 pb-28 pt-4 sm:max-w-md">
          <div className="flex pb-4">
            <FreedomlifeIcon className="w-[230px]" />
          </div>

          <GuideMonthButtonWrapper />

          <GuideList
            queryData={queryData}
            getGuideHasBeenRead={getGuideHasBeenRead}
            onGuideClick={onGuideClick}
            onCheckMarkClick={onCheckMarkClick}
          />
        </div>

        <GuideMonthModal />
      </GuideModalsContextProvider>
    </>
  )
}

function GuideMonthButtonWrapper() {
  const { setOpenGuideMonth } = useGuideModalsContext()

  return (
    <GuideMonthButton
      redirectToGuideMonthScreen={() => setOpenGuideMonth(true)}
    />
  )
}

function GuideMonthModal() {
  const { openGuideMonth, setOpenGuideMonth } = useGuideModalsContext()
  const selectedGuideMonth = useReadPassageGeneralContext(
    (state) => state.selectedGuideMonth,
  )
  const { setSelectedGuideMonth } = useReadPassageGeneralContext(
    (state) => state.actions,
  )

  const onMonthClick = (monthString: string) => {
    setSelectedGuideMonth(monthString)
    setOpenGuideMonth(false)
  }

  return (
    <Drawer
      open={openGuideMonth}
      title="Pilih Bulan Panduan"
      setOpen={setOpenGuideMonth}
    >
      <GuideMonthList
        selectedGuideMonth={selectedGuideMonth}
        onMonthClick={onMonthClick}
      />
    </Drawer>
  )
}
