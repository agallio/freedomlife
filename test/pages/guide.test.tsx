import { render } from '../testUtils'
import { Guide } from '../../pages/guide'

describe('Guide Page', () => {
  describe('Header', () => {
    // it('title rendered as "Panduan Baca"', () => {
    //   const { getByTestId } = render(<Guide />, {})
    //   const headerTitle = getByTestId('headerTitle')
    //   expect(headerTitle.textContent).toBe('Panduan Baca')
    // })

    // Christmas Edition
    it('title rendered as "Panduan Baca 🎁"', () => {
      const { getByTestId } = render(<Guide />, {})
      const headerTitle = getByTestId('headerTitle')
      expect(headerTitle.textContent).toBe('Panduan Baca 🎁')
    })

    it('subtitle rendered as "Panduan Baca Bulan Ini"', () => {
      const { getByTestId } = render(<Guide />, {})
      const headerSubtitle = getByTestId('headerSubtitle')
      expect(headerSubtitle.textContent).toBe('Panduan Baca Bulan Ini')
    })
  })
})
