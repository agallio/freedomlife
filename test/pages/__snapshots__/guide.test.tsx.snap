// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guide Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    __source="[object Object]"
    class="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-indeterminate"
    role="progressbar"
  >
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate"
    />
    <div
      class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorSecondary"
    />
  </div>
  <div
    __source="[object Object]"
    class="container"
    style="padding-bottom: 25px; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <h2
      __source="[object Object]"
      class="header__title"
      data-testid="headerTitle"
      style="margin-top: 65px;"
    >
      Panduan Baca
    </h2>
    <p
      __source="[object Object]"
      class="header__subtitle"
      data-testid="headerSubtitle"
    >
      Panduan Baca Bulan Ini
    </p>
  </div>
  <div
    __source="[object Object]"
    class="container--fluid"
    style="padding-bottom: 55px;"
  >
    <div
      __source="[object Object]"
      class="MuiPaper-root MuiCard-root guidecard MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        __source="[object Object]"
        class="MuiCardContent-root"
        style="padding: 16px 10%; height: 80vh;"
      >
        <div
          __source="[object Object]"
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
        >
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
              style="width: 105px; height: 90px; border-radius: 5px;"
            />
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
          </div>
        </div>
        <div
          __source="[object Object]"
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
        >
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
              style="width: 105px; height: 90px; border-radius: 5px;"
            />
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
          </div>
        </div>
        <div
          __source="[object Object]"
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
        >
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
              style="width: 105px; height: 90px; border-radius: 5px;"
            />
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
          </div>
        </div>
        <div
          __source="[object Object]"
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
        >
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
              style="width: 105px; height: 90px; border-radius: 5px;"
            />
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
          </div>
        </div>
        <div
          __source="[object Object]"
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
        >
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
              style="width: 105px; height: 90px; border-radius: 5px;"
            />
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
          </div>
        </div>
        <div
          __source="[object Object]"
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
        >
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
              style="width: 105px; height: 90px; border-radius: 5px;"
            />
          </div>
          <div
            __source="[object Object]"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-8"
          >
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
            <span
              __source="[object Object]"
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
              style="width: 150px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
